qiskit/chemistry/drivers/fcidumpd/dumper.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+""" FCIDump dumper """
+
+from io import TextIOWrapper
+import itertools
+import numpy as np
+
+
+def dump(outpath: str, norb: int, nelec: int, hijs: list, hijkls: list, einact: float,
+         ms2: int = 0, orbsym: list = None, isym: int = 1) -> None:
+    """ Generates a FCIDump output.
+        Args:
+            outpath: path to the output file
+            norb: number of orbitals
+            nelec: number of electrons
+            hijs: pair of alpha and beta 1-electron integrals. The latter may be None.
+            hijkls: triplet of alpha/alpha, beta/alpha and beta/beta 2-electron integrals. The
+            latter two may be None.
+            einact: inactive energy
+            ms2 (optional): 2*S where S is the spin quantum number. Defaults to 0.
+            orbsym (optional): list of spatial symmetries of the orbitals
+            isym (optional): spatial symmetry of the wave function. Defaults to 1.
+    """
```

Comment:
* can you use `List[element type]` instead of `list`?
* optional is also specified in the signature as `orbsym: Optional[List[element type]] = None` and not in the docstring :slightly_smiling_face: 

These types are defined in typing, i.e. add `from typing import List, Optional` as first import.